src/utils/utilities.jl

```diff
@@ -454,3 +454,45 @@ function isdivisible(
     # have any common divisor)
     return isinteger(Dates.Day(1) / dt_small)
 end
+
+"""
+    promote_period(period::Dates.Period)
+
+Promote a period to the largest possible period type.
+
+This function attempts to represent a given `Period` using the largest possible
+unit of time. For example, a period of 24 hours will be promoted to 1 day.
+
+# Examples
+```julia-repl
+julia> promote_period(Hour(24))
+1 day
+
+julia> promote_period(Day(14))
+2 weeks
+
+julia> promote_period(Day(365))
+1 year
+
+julia> promote_period(Millisecond(1))
+1 millisecond
+```
+"""
+function promote_period(period::Dates.Period)
```

Comment:
Should we maybe call this `round_period`? Promote kind of sounds like the value is not changing, but it may (even if it's just approximate)

Reply:
Indeed, the value doesn't change. This "promotes" only when it is possible to use a single unit. For example, if you try promoting 86401 seconds, you will get back 86401 seconds (but if you promote 86400 seconds, you get 1 day). To me `round` implies that there might be loss of precision. 

I don't know what better word to use.

Reply:
Yeah, `promote_period(Day(365)) => 1 year` seems like it's rounding

Reply:
Ah, right, I removed that because indeed it was not possible without rounding! Thanks for catching that.